skyjo.py

```diff
@@ -285,39 +355,95 @@ def get_is_visible(skyjo: Skyjo, player: int = 0) -> bool:
 
     table = skyjo[1]
 
-    return table[player, :, :, FINGER_HIDDEN].any()
+    return _player_table_is_visible(table, player)
 
 
 def get_score(skyjo: Skyjo, player: int = 0) -> int:
     """Get the score of visible cards on a player's board."""
 
     table = skyjo[1]
 
-    score = 0
-    for row in range(ROW_COUNT):
-        for column in range(COLUMN_COUNT):
-            for card in range(CARD_SIZE):  # Skip CLEARED, score zero
-                if table[player, row, column, card]:
-                    score += card - 2
-                    break
+    return (
+        (
+            # get all card index of card ignoring cleared and face-down
+            np.argwhere(table[player, :, :, :15] == 1)[:, 2]
+            - 2  # since card index starts with -2
+        )
+        .sum()
+        .item()
+    )
+
+
+def get_round_scores(
+    skyjo: Skyjo, round_ending_player: int = 0
+) -> np.ndarray[tuple[int], np.uint8]:
+    """Get the scores of all players for the current round."""
+    players = skyjo[3]
+    base_scores = np.array([get_score(skyjo, player=i) for i in range(players)])
+    round_ender_score = base_scores[round_ending_player]
+    base_scores[round_ending_player] = 1000  # larger than any other possible score
```

Comment:
more foolproof to just use `inf`? feels like there should be a cooler way to do this :(